### PR TITLE
btrfs-dedupe: add new package at the latest git revision.

### PR DESCRIPTION
**Pull request by ikervagyok**
still marked broken, as i don't have experience with either rust nor btrfs-dedupe. also i have no throw-away partition atm, to test this - please test and remove the broken flag as soon as it is confirmed working

###### Things done
since i didn't find any direct cargo-to-nix transition, i took pijul
as a template. also, the `crates.io`-version is 1.2, this is 1.3
plus a few later patches.

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by ikervagyok** [comment]
@joachifm it's not really WIP - it's marked with `meta.broken-true;` in the commit, so anyone testing it has to be aware of it. except of not being able to test it's primary function (lacking a workstation with vm's and throwaway data) it should work, it prints its help mesage just fine.
i think it probably needs these labels:
- 6.`topic: rust`
- 8.`has: package (new)`
- 9.`needs: community feedback` (or rather a new one: `needs: testing`)

**Comment by joachifm** [comment]
As far as I'm concerned, it's wip until confirmed as working. Adding new stuff already marked as broken doesn't make any sense to me.

**Comment by ikervagyok** [comment]
ok, that seems reasonable.
then i'll remove the `meta.broken` attribute ;)

File: pkgs/tools/filesystems/btrfs-dedupe/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchgit, rustPlatform, lzo, zlib }:
+
+with rustPlatform;
+
+buildRustPackage rec {
+  name = "btrfs-dedupe-${version}";
+  version = "1.0.3-git";
```
Comment (joachifm) [inline pkgs/tools/filesystems/btrfs-dedupe/default.nix]: Why not `1.1.0`?

File: pkgs/tools/filesystems/btrfs-dedupe/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  src = fetchgit {
```
Comment (joachifm) [inline pkgs/tools/filesystems/btrfs-dedupe/default.nix]: Does this rely on submodules or must cargo builds use git checkouts? If not, using source/release archives is preferred (better caching). `fetchFromGitHub` is a convenient method to do this.

**Comment by joachifm** [comment]
If the build succeeds and there's nothing about the build procedure itself that would cause the resulting artefact to misbehave, it makes sene to not mark it as broken, imo; any issues would then be upstream issues. That said, for something like this it'd be good if users could trust that it had been verified to actually work.

**Pull request merged**